src/main/kotlin/org/opensearch/indexmanagement/snapshotmanagement/model/SMPolicy.kt

```
@@ -155,35 +159,42 @@ data class SMPolicy(
             }
 
             if (enabled && enabledTime == null) {
-                enabledTime = Instant.now()
+                enabledTime = now()
             } else if (!enabled) {
                 enabledTime = null
             }
 
-            // TODO SM update policy API can update this value
             if (lastUpdatedTime == null) {
-                lastUpdatedTime = Instant.now()
+                lastUpdatedTime = now()
             }
 
             if (schedule == null) {
-                schedule = IntervalSchedule(Instant.now(), 1, ChronoUnit.MINUTES)
+                schedule = IntervalSchedule(now(), 1, ChronoUnit.MINUTES)
             }
 
-            require(creation != null) { "Must provide the schedule for snapshot creation." }
+            require(creation != null) { "Must provide the creation configuration." }
             // If user doesn't provide delete schedule, use the creation schedule
             if (deletion != null && !deletion.scheduleProvided) {
                 deletion = deletion.copy(
                     schedule = creation.schedule
                 )
             }
 
+            requireNotNull(snapshotConfig) { "$SNAPSHOT_CONFIG_FIELD field must not be null" }
+            if (snapshotConfig[DATE_FORMAT_FIELD] != null) {
+                val validMsg = validateDateFormat(snapshotConfig[DATE_FORMAT_FIELD] as String)
+                if (validMsg != null) {
+                    throw IllegalArgumentException("Please provide a valid date format. $validMsg")
+                }
```

Feels like we should just throw the exception from validateDateFormat